app/services/katello/ui_notifications/pulp/proxy_disk_space.rb

```diff
@@ -0,0 +1,39 @@
+module Katello
+  module UINotifications
+    module Pulp
+      class ProxyDiskSpace
+        class << self
+          def deliver!
+            SmartProxy.unscoped.with_features('Pulp').each do |proxy|
+              percentage = proxy.statuses[:pulp].storage['pulp_dir']['percent']
+              next unless percentage[0..2].to_i > 90
+              next if notification_already_exists?(proxy)
```

Comment:
I would think this logic is better (saves on some sql's)
https://github.com/theforeman/foreman/blob/develop/app/services/ui_notifications/hosts/build_completed.rb#L7